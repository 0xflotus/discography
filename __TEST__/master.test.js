import testSnapshot from './fixtures/testSnapshot';

test('Should have all basic fields on the master type', async () => {
  const query = `
    query {
      lookup {
        master (id: 18500) {
          id
          title
          year
          genres
          styles
        }
      }
    }
  `;

  await testSnapshot(query);
});

test('Should implement the master-release connection', async () => {
  const query = `
    query {
      lookup {
        master (id: 18500) {
          releases (first: 1) {
            edges {
              node {
                id
                title
              }
            }
          }
        }
      }
    }
  `;

  await testSnapshot(query);
});
