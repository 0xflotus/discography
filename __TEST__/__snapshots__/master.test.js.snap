// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should have all basic fields on the master type 1`] = `
Object {
  "data": Object {
    "lookup": Object {
      "master": Object {
        "genres": Array [
          "Electronic",
        ],
        "id": 18500,
        "styles": Array [
          "Techno",
        ],
        "title": "New Soil",
        "year": 2001,
      },
    },
  },
}
`;

exports[`Should implement the master-release connection 1`] = `
Object {
  "data": Object {
    "lookup": Object {
      "master": Object {
        "releases": Object {
          "edges": Array [
            Object {
              "node": Object {
                "id": 5477,
                "title": "New Soil",
              },
            },
          ],
        },
      },
    },
  },
}
`;
